src/modules/dpos/validator/hooks/useDelegateName.js

```diff
@@ -30,39 +29,44 @@ const useDelegateName = (value) => {
   });
   const timeout = useRef();
   const { t } = useTranslation();
-  const network = useSelector(state => state.network);
-
-  const checkUsername = () => {
-    clearTimeout(timeout.current);
+  const [params, setParams] = useState({});
+  const { data, isLoading, error } = useDelegates({
+    config: { params },
+    options: { enabled: name.value?.length >= 3, retry: false },
+  });
 
-    timeout.current = setTimeout(() => {
+  const checkUsername = async () => {
+    if (name.value.length >= 3 && name.value.length <= 20 && isLoading) {
       setName({
         ...name,
         loading: true,
       });
-      getDelegate({ network, params: { username: name.value } })
-        .then(() => {
-          setName({
-            ...name,
-            loading: false,
-            error: true,
-            message: t('"{{username}}" is already taken.', { username: name.value }),
-          });
-        })
-        .catch(() => {
-          setName({
-            ...name,
-            error: false,
-            loading: false,
-          });
+    }
+    if (name.value.length >= 3 && !isLoading) {
+      if (error) {
+        setName({
+          ...name,
+          error: false,
+          loading: false,
+        });
+      } else {
+        setName({
+          ...name,
+          loading: false,
+          error: true,
+          message: t('"{{username}}" is already taken.', { username: name.value }),
         });
-    }, 1000);
+      }
+    }
   };
 
   useEffect(() => {
+    clearTimeout(timeout.current);
     const formatError = validate(name.value, t);
     if (!formatError) {
-      checkUsername();
+      timeout.current = setTimeout(() => {
+        setParams({ name: name.value });
+      }, 500);
```

Comment:
Seems as though we have decreased the debounce time from 1sec to 500 ms. Was it intentional?